gems/pending/Scvmm/miq_hyperv_disk.rb

```diff
@@ -118,27 +118,34 @@ def bread_cached(start_sector, number_sectors)
   end
 
   def bread(start_sector, number_sectors)
-    $log.debug "miq_hyperv_disk.bread(#{start_sector}, #{number_sectors})"
+    log_header = "MIQ(#{self.class.name}.#{__method__}:"
+    $log.debug "#{log_header} (#{start_sector}, #{number_sectors})"
     return nil if start_sector > @lba_end
     number_sectors = @size_in_blocks - start_sector if (start_sector + number_sectors) > @size_in_blocks
+    expected_bytes = number_sectors * @block_size
+    retries        = 0
     read_script = <<-READ_EOL
 $file_stream = [System.IO.File]::Open("#{@virtual_disk}", "Open", "Read", "Read")
 $buffer      = New-Object System.Byte[] #{number_sectors * @block_size}
 $file_stream.seek(#{start_sector * @block_size}, 0)
-$file_stream.read($buffer, 0, #{number_sectors * @block_size})
+$file_stream.read($buffer, 0, #{expected_bytes})
 [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes($buffer))
 $file_stream.Close()
 READ_EOL
 
-    # TODO: Error Handling
-    t1           = Time.now.getlocal
-    encoded_data = @parser.output_to_attribute(@winrm.run_powershell_script(read_script))
-    t2           = Time.now.getlocal
-    buffer       = ""
-    Base64.decode64(encoded_data).split(' ').each { |c| buffer += c.to_i.chr }
-    @total_copy_from_remote_time += t2 - t1
-    @total_read_execution_time += Time.now.getlocal - t1
-    buffer
+    until retries == BREAD_RETRIES
```

Comment:
You could make this;
```
(0...BREAD_RETRIES).each do
```
Then, you wouldn't need to increment and compare the `retries` variable.